proxies/shared/resources/jsc/SetBackendCorrelationId.js

```diff
@@ -1,8 +1,9 @@
 const incomingCorrelationId = context.getVariable("request.header.x-correlation-id");
 const messageId = context.getVariable("messageid");
 
-if (incomingCorrelationId) {
-    context.setVariable("backendCorrelationId", incomingCorrelationId + "." + messageId);
-} else {
+if (incomingCorrelationId && messageId) {
+    context.setVariable("backendCorrelationId", incomingCorrelationId);
+}
+else {
     context.setVariable("backendCorrelationId", messageId);
 }
```

Comment:
Is there a reason why `messageId` is tested?

```suggestion
if (incomingCorrelationId) {
    context.setVariable("backendCorrelationId", incomingCorrelationId);
}
else {
    context.setVariable("backendCorrelationId", messageId);
}
```